File: day-1/java/Calories.java

import java.io.File;
import java.util.Scanner;



public class Calories {
    
    public static void main(String[] args) throws Exception {
        File file = new File("inputs.txt");
        Scanner sc = new Scanner(file);
        int highest_1 = 0;
        int highest_2 = 0;
        int highest_3 = 0;
        int sum = 0;
        
        while(sc.hasNextLine()){
            String line = sc.nextLine();
            if(line.isEmpty()){
                if(sum > highest_1){
                    highest_3 = highest_2;
                    highest_2 = highest_1;
                    highest_1 = sum;
                } else if(sum > highest_2){
                    highest_3 = highest_2;
                    highest_2 = sum;
                } else if(sum > highest_3){
                    highest_3 = sum;
                }
                sum = 0;
                continue;
            }
            int n = Integer.parseInt(line);
            sum += n;
        }
        sc.close();
        if(sum > highest_1){
            highest_3 = highest_2;
            highest_2 = highest_1;
            highest_1 = sum;
        } else if(sum > highest_2){
            highest_3 = highest_2;
            highest_2 = sum;
        } else if(sum > highest_3){
            highest_3 = sum;
        }

        System.out.println("Highest is = " + highest_1);
        System.out.println("Total is = " + (highest_1 + highest_2 + highest_3));
    }
    
}

File: day-1/go/calories.go
package main

import (
	"fmt"
	"bufio"
	"os"
	"strconv"
)

func main(){
	var highest_1 = 0;
	var highest_2 = 0;
	var highest_3 = 0;
	var sum = 0;

	file, err := os.Open("inputs.txt")
	if err != nil {
		fmt.Println("input file not found");
		return;
	}
	defer file.Close()
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		var line = scanner.Text()
		if line == "" {
			if sum > highest_1 {
				highest_3 = highest_2
				highest_2 = highest_1
				highest_1 = sum
			} else if sum > highest_2 {
				highest_3 = highest_2
				highest_2 = sum
			} else if sum > highest_3 {
				highest_3 = sum
			}
			sum = 0
			continue
		}
		num, err := strconv.Atoi(line)
		if err != nil {
			fmt.Println("not able to convert value to int")
			return
		}
        sum += num
    }
	if sum > highest_1 {
		highest_3 = highest_2
		highest_2 = highest_1
		highest_1 = sum
	} else if sum > highest_2 {
		highest_3 = highest_2
		highest_2 = sum
	} else if sum > highest_3 {
		highest_3 = sum
	}
	fmt.Println("Highest is: ", highest_1);
	fmt.Println("Total of highest 3: ", highest_1 + highest_2 + highest_3);
}